app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PaymentsController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  load_resource :conference, find_by: :short_title
+  authorize_resource :conference_registrations, class: Registration
+
+  def index
+    @payments = current_user.payments
+  end
+
+  def new
+    @total_amount_to_pay = Ticket.total_price(@conference, current_user, paid: false)
+    @unpaid_ticket_purchases = current_user.ticket_purchases.unpaid.by_conference(@conference)
+  end
+
+  def create
+    @payment = Payment.new payment_params
+
+    if @payment.purchase && @payment.save
+      update_purchased_ticket_purchases
+      redirect_to conference_conference_registration_path(@conference.short_title),
+                  notice: 'Thanks! You have purchased your tickets successfully.'
+    else
+      @total_amount_to_pay = Ticket.total_price(@conference, current_user, paid: false)
```

Comment:
Why don't we show the error in a flash message?
In Payment::purchase we add StripeError to ``errors.add(:base, error.message)`` so we should also propagate it to the users, shouldn't we?